cms/djangoapps/contentstore/views/course.py

```diff
@@ -1600,20 +1698,42 @@ def group_configurations_detail_handler(request, course_key_string, group_config
             store.update_item(course, request.user.id)
             configuration = GroupConfiguration.update_usage_info(store, course, new_configuration)
             return JsonResponse(configuration, status=201)
+
         elif request.method == "DELETE":
             if not configuration:
                 return JsonResponse(status=404)
 
             # Verify that group configuration is not already in use.
-            usages = GroupConfiguration.get_usage_info(course, store)
-            if usages.get(int(group_configuration_id)):
-                return JsonResponse(
-                    {"error": _("This Group Configuration is already in use and cannot be removed.")},
-                    status=400
-                )
-
             index = course.user_partitions.index(configuration)
-            course.user_partitions.pop(index)
+            if configuration.scheme.name == 'random':
+                usages = GroupConfiguration.get_usage_info(course, store)
+                used = True if usages.get(int(group_configuration_id)) else False
+
+                if used:
+                    return JsonResponse(
+                        {"error": _("This group configuration is in use and cannot be deleted.")},
+                        status=400
+                    )
+                course.user_partitions.pop(index)
+            else:
+                if not group_id:
+                    return JsonResponse(status=404)
+
+                group_id = int(group_id)
+                usages = GroupConfiguration.get_content_groups_usage_info(store, course)
+                used = True if usages.get(group_id) else False
+
+                if used:
+                    return JsonResponse(
+                        {"error": _("This content group is in use and cannot be deleted.")},
+                        status=400
+                    )
+                configuration = GroupConfiguration.remove_content_group_from_configuration(configuration, group_id)
```

Comment:
If the delete logic is in a separate method, then I don't think you need to make this into a separate method since it is only called from here.